crates/iceberg/src/expr/predicate.rs

```diff
@@ -879,12 +905,19 @@ mod tests {
         )
     }
 
+    fn test_bound_predicate_serialize_diserialize(bound_predicate: BoundPredicate) {
```

Comment:
https://github.com/apache/iceberg-rust/blob/ea8d72924bf4da04a70b8aa3bb553a0e168e747b/crates/iceberg/src/expr/predicate.rs#L1200
case like this will test it?